### ISSUE TITLE
setOptions autoScroll to false scrolls to beginning

### ISSUE THREAD

**Issue by larsrgit** [katspaugh/wavesurfer.js#3189]
## Bug description
When setting autoScroll to false with setOptions, the view scrolls to the beginning of the audio

## Environment
 - Browser: Chrome

## Minimal code snippet
Easy to test by just adding a checkbox for autoScroll in the zoom demo

## Expected result
Auto scroll is disabled and the view does not scroll

## Obtained result
Auto scroll is disabled and view scrolls to the beginning of the audio.

**Comment by larsrgit** [comment]
The same happens when zooming. When zooming with playback paused, the view scrolls to the beginning, even with autoCenter enabled. In my opinion, the intuitive behaviour would be to zoom into the current view position.

**Comment by katspaugh** [comment]
I agree. Please feel free to work on it, I'm there for any questions.

**Comment by larsrgit** [comment]
It seems that evertime reRender is called, the position is set to the beginning. However I'm lost on why this happens. The code in reRender looks good for adjusting the view back to the cursor assuming the view was at the cursor. This might not work, when we the view is not at the cursor to begin with, but does not explain why it jumps to the beginning. I could not find anything else where the scrollposition is changed during rerendering.

For now I just added 

`const { scrollLeft, scrollWidth, clientWidth } = wavesurfer.renderer.scrollContainer `
`const curentRelPos = (scrollLeft + clientWidth * 0.5) / scrollWidth`

before zooming and 

` wavesurfer.renderer.scrollContainer.scrollLeft = curentRelPos * wavesurfer.renderer.scrollContainer.scrollWidth - clientWidth / 2 `

after zooming

**Comment by larsrgit** [comment]
However as I understand the renderChannel method, it starts rendering at the current scroll position. So with my quick fix on my side it still always starts rendering at the beginning. So when changing the scroll after rendering back to the previous position it takes quite some time to render at that position if that position is at a late position of a longer audio. This is related to #3193 and makes this feel very unresponsive.

**Comment by katspaugh** [comment]
I think it might be jumping because it deletes all the canvases, so there's no scroll until it renders new canvases? Not sure if the wrapper has a width.

**Comment by larsrgit** [comment]
Unfortunately this question is beyond my very limited Javascript skills
